velox/common/memory/SharedArbitrator.h

```diff
@@ -400,21 +427,25 @@ class SharedArbitrator : public memory::MemoryArbitrator {
   int64_t minGrowCapacity(const MemoryPool& pool) const;
 
   // Returns true if 'pool' is under memory arbitration.
-  bool isUnderArbitration(MemoryPool* pool) const;
   bool isUnderArbitrationLocked(MemoryPool* pool) const;
 
   void updateArbitrationRequestStats();
+
   void updateArbitrationFailureStats();
 
   const uint64_t reservedCapacity_;
+  const uint64_t memoryPoolInitialCapacity_;
   const uint64_t memoryPoolReservedCapacity_;
   const uint64_t memoryPoolTransferCapacity_;
   const uint64_t memoryReclaimWaitMs_;
   const bool globalArbitrationEnabled_;
   const bool checkUsageLeak_;
 
+  mutable folly::SharedMutex poolLock_;
+  std::unordered_map<MemoryPool*, std::weak_ptr<MemoryPool>> candidates_;
```

Comment:
Could we use

folly::Synchronized<std::unordered_map<MemoryPool*, std::weak_ptr<MemoryPool>, folly::SharedMutex> candidates_;

Since there are multiple mutex in this class. It will be easier to manage and less error prone. The effect will be the same, too.

Reply:
This is under change in followup. I prefer explicit lock management.